Lib/test/test_typing.py

```diff
@@ -4920,6 +4920,40 @@ class B(Generic[S]): ...
         class C(List[int], B): ...
         self.assertEqual(C.__mro__, (C, list, B, Generic, object))
 
+    def test_multiple_inheritance_with_non_type_implementing___mro_entries__(self):
+        class O:
+            def __mro_entries__(self, bases):
+                return (object,)
+
+        class A(List[int], O()): ...
+
+        self.assertEqual(A.__mro__, (A, list, Generic, object))
+
+    def test_multiple_inheritance_with_non_type_without___mro_entries__(self):
+        class O: pass
+
+        # We should get an error here, but from the type machinery, not from typing.py
+        with self.assertRaisesRegex(TypeError, r"^metaclass conflict:"):
+            class A(List[int], O()): ...
+
+    def test_multiple_inheritance_with_non_type_with_bad___mro_entries__(self):
+        class O:
+            def __mro_entries__(self, bases):
+                return None
+
+        # We should get an error here, but from the type machinery, not from typing.py
+        with self.assertRaisesRegex(TypeError, r"^__mro_entries__ must return a tuple"):
+            class A(List[int], O()): ...
```

Comment:
```suggestion
        class ImplementsBadMroEntries:
            def __mro_entries__(self, bases):
                return None

        # We should get an error here, but from the type machinery, not from typing.py
        with self.assertRaisesRegex(TypeError, r"^__mro_entries__ must return a tuple"):
            class A(List[int], ImplementsBadMroEntries()): ...
```

Reply:
I am not sure that long and hard to read name is better that a short name, unique in the test method. But may be it is only me, I have a physical issue with reading long things.

Reply:
I prefer tests to be self-documenting where possible; this is how I would write it

Reply:
As I say, it's just my personal preference.

Reply:
I tried to thread this needle by using more meaningful names, but still shorter by eliminating the common prefix: `GoodEntries` and `BadEntries`.